Kitodo-API/src/main/java/org/kitodo/api/dataformat/SortedCollection.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.api.dataformat;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.ListIterator;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * A collection that is aware of and enforces the order of its members
+ * prescribed by the ordinal number of the media units.
+ */
+class SortedCollection<T> extends ArrayList<T> implements Collection<T> {
```

Comment:
Why don't you use a set? If you overwrite the compare method of a set, the elements are inserted at the correct position, you don't need to block the methods then.
When someone uses a for loop for this sortedCollection, then it's blocked, and it is not obvious why.